haskell-style.md

```diff
@@ -588,12 +588,29 @@ module Driver
 
 ## Declaring Extensions
 
-- Declare extensions in the module(s) that use them, i.e. don't use
-  `default-extensions`
+- All Haskell packages MUST use the following `default-extensions`:
+
+  ```yaml
+  default-extensions:
+    - BangPatterns
+    - DeriveGeneric
+    - LambdaCase
+    - NoImplicitPrelude
+    - NoMonomorphismRestriction
+    - OverloadedStrings
+    - RankNTypes
+    - RecordWildCards
+    - ScopedTypeVariables
+    - TypeApplications
+    - TypeFamilies
+  ```
+
+  **NOTE**: `NoImplicitPrelude` may be omitted in packages using the normal,
```

Comment:
Does this jive with tooling?

Reply:
I think it does, since it doesn't have any syntactic effects.

Reply:
I think you're right in that if a syntax-only tool has `NoImplicitPrelude` in its own extensions configuration, but a package does not (and so doesn't `import Prelude`) it still works because it's syntax-only. So I think we're good.

I was worried for a second about DocTest, but since you still define those extensions in the executable in the package itself, you can just make it consistent either way. Similar story for a package-local `.ghci`.